### PR TITLE
Accept cmdstanvb, draws_array or draws_matrix as fitted_params

### LINKED ISSUE #387: Allow $generate_quantities() to directly use draws_array objects
Mentioned by @rok-cesnovar at https://discourse.mc-stan.org/t/should-generate-quantities-accept-cmdstanvb-objects/19628. This would allow us, for example, to use variational Bayes draws to run generated quantities.

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Fixes #387 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar, Uni. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: R/data.R
Comment (jgabry) [inline R/data.R:149]: This looks good, but I think there's a slightly nicer way code it using the `posterior::draws_array()` constructor. I can do that in a few minutes and push it.

Comment (rok-cesnovar) [reply R/data.R:149]: That would be fantastic! Thanks!

File: R/data.R
Comment (jgabry) [inline R/data.R:165]: Does it make sense to use `data.table::fwrite()` here to write the files instead of `utils::write.table()`? I'm not sure how big of a difference the speed is for writing, but just asking since we're already using `data.table::fread()`. We could also just use write.table() for now and potentially check in the future if it makes sense to switch (we don't necessarily need to hold up this PR).

Comment (rok-cesnovar) [reply R/data.R:165]: We can yeah, good call. I think the API is the same.

File: R/data.R
Comment (jgabry) [reply R/data.R]: Just pushed it. Let me know if you like it. If not we can revert because what you had definitely also worked.

File: R/data.R
Comment (rok-cesnovar) [reply R/data.R:171]: Turns out it's not as straightforward. Will make an issue. I doubt it makes a huge difference though.

**Comment by jgabry** [comment]
I also just update the doc for the fitted_params argument.

**Comment by rok-cesnovar** [comment]
Oh right, thanks!

File: R/data.R
Comment (jgabry) [reply R/data.R:171]: Ok no problem. We can use write.table().

**Comment by jgabry** [comment]
I didn't test this on any of my own models but the tests you wrote seem like they cover all the cases, so if you like the coding changes I made to `draws_to_csv()` then I think this is probably ready to go.

File: R/data.R
Comment (rok-cesnovar) [reply R/data.R]: Yes, I like it.

**Comment by rok-cesnovar** [comment]
Everything looks much nicer! Thanks. Will merge when tests pass.

**Review by jgabry — APPROVED** [review]

**Comment by jgabry** [comment]
Looks like all checks are passing so I'm going to go ahead and merge this.

**Pull request merged**
